src/app/pages/wallets/wallet-details/transaction-editor/pages-wallet-transaction-editor.component.ts

```diff
@@ -105,4 +122,17 @@ export class PagesWalletTransactionEditorComponent implements OnInit {
       return { amountIsZero: true };
     };
   }
+
+  private getCategoriesByType(typeToFilter: WalletTransactionType): void {
+    this.transactionCategoriesService.getCategories().subscribe((categoriesReceived) => {
```

Comment:
You are not handling failed requests, the flow should be I think:
- open editor
- show loading indicator (do not show form yet)
- if request will fail, close editor and show message "Editing transaction failed"
- if request will not fail, create and show the form and hide loading indicator

Besides that I would separate getting all categories from filtering them as now because of that you filter them both in `getCategoriesByType` and `listetoTypeChange` - not needed duplication of code.